third_party/nvfuser/csrc/multidevice/aggregate_dag.cpp

```diff
@@ -0,0 +1,211 @@
+#include <multidevice/aggregate_dag.h>
+#include <multidevice/multigroup_fusion.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+AggregateDag::AggregateDag() : Fusion() {}
+
+void AggregateDag::build(MultiGroupFusion* fusion) {
+  // iterate over all groups in the multigroup fusion
+  for (auto group : fusion->groups()) {
+    // set the container we will add the IR to
+    auto container = reinterpret_cast<IrContainer*>(this);    // set the container we will add the IR to
```

Comment:
Is this `reinterpret_cast` necessary? `AggregateDag` is a `IrContainer`, right?

Reply:
replaced by `->as<IrContainer>`. But without it, it doesn't compile for some reason

Reply:
solved

Reply:
I think that the problem comes from overload resolution and ellipsis. In IrBuilder class we have two overloaded functions:
`static T* create(Args&&... args)`
`static T* create(IrContainer* container, Args&&... args)`
what we want is to call the later one. But if I don't explicitly upscast AggregateDag to IrContainer the first one is selected, and we get a compile error. 

So for now I leave `this->as<IrContainer>`, but @naoyam let me know if you have a better solution.

Reply:
Yes, indeed the `as()` method used here calls `dynamic_cast` or `static_cast`